### PR TITLE
Dossier

### PR DESCRIPTION
**Pull request by lukasbuenger**

### CODE REVIEW

**Comment by lukasbuenger** [comment]
Damn. I'll get to it

**Comment by lukasbuenger** [comment]
Ok, I checked it out. There's the following options:
1. Disable UI, achievable through schema, add  `showUI: false` to editorOptions of tile teasers. Adding and removing teasers works through button and backspace
2. Put the UI in the teaser. Would need to implement a custom inline ui for just article tiles.
For everything else we'd have to touch the styleguide components.

**Comment by tpreusse** [comment]
@lukasbuenger where are those portals when we need em ;) Out of those two I'd say number 1. Other solution come to mind, like not rendering the UI when `isSelected` `false` but ultimately wouldn't be clean either. It would be nice to tackle the whole UI refactoring soon

**Comment by tpreusse** [comment]
@lukasbuenger the naming is in-consistent in the article schema

Zone: TEASER type: articleDossier
Module: articleDossier
`modules/dossier/article`

 - Heading level: 2
   Module: headline
  `modules/headline`
 - Zone: TEASERGROUP
    Module: articleCollection
    `modules/articlecollection`
   - Zone: TEASER type: "articleTile"
      Module: teaser
      `modules/teaser`

---

I'd prefer some like this:

Zone: ARTICLECOLLECTION
Module: articleCollection
`modules/article/collection`

 - Heading level: 2
   Module: headline
  `modules/headline`
 - Zone: TEASERGROUP
    Module: articleGroup
    `modules/article/group`
   - Zone: TEASER type: "articleTile"
      Module: teaser
      `modules/teaser`
  

Generally the naming in the article schema should be generic. It will not only be used for dossiers.

**Comment by lukasbuenger** [comment]
@tpreusse Check again. Includes `membersOnly` key on ARTICLECOLLECTION zone. See also updated orbiting/styleguide#dossier-v2

**Comment by tpreusse** [comment]
Awesome, thank you. What the default value of `membersOnly`? Should be `true` on inserting a new collection. Could we add an additional text field `unauthorizedText` («Platzhalt für nicht Mitglieder») I'll show a default text if not set but would be nice to say some like «Die Hintergründe analysiert in 5 Stücken für unsere Mitglieder.» etc.

**Comment by lukasbuenger** [comment]
Currently is true. Will add the text field.

**Comment by lukasbuenger** [comment]
@tpreusse I added a field that points at  `unauthorizedText` in the `ARTICLECOLLECTION` zone

File: components/editor/modules/article/collection.js [hunk 1]
```diff
@@ -151,6 +153,12 @@ export const articleCollectionForm = options => {
     >
       Nur für Members sichtbar?
     </Checkbox>
+      <Field
+        label='Nachricht für Nicht-Members'
+        value={articleCollection.data.get('unauthorizedText')}
+        onChange={(_, text) => onChange(
+          value.change().setNodeByKey(articleCollection.key, { data: { unauthorizedText: text } })
```
Comment (tpreusse) [inline components/editor/modules/article/collection.js]: should be `{ data: articleCollection.data.set('unauthorizedText', text) }`

File: components/editor/modules/article/collection.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    )
+    if (!articleCollection) {
+      return null
+    }
+    return <UIForm>
+      <Label>Artikelsammlung</Label>
+      <Checkbox
+        checked={articleCollection.data.get('membersOnly')}
+        onChange={(_, checked) => onChange(
+          value.change().setNodeByKey(articleCollection.key, { data: { membersOnly: checked } })
```
Comment (tpreusse) [inline components/editor/modules/article/collection.js]: `{ data: articleCollection.data.set('membersOnly', checked) }`

File: components/editor/modules/article/collection.js [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+      }),
+      articleGroupModule.helpers.newItem()
+    ]
+  })
+}
+
+export const getData = data => ({
+  membersOnly: true,
+  unauthorizedText: '',
+  ...(data || {})
```
Comment (tpreusse) [inline components/editor/modules/article/collection.js]: nit: `|| {}` not needed, spreading undefined is supported (behaviour of object spread is the same as `Object.assign`)

**Pull request merged**